### PR TITLE
Height test ratio status

### PR DESCRIPTION
**Pull request by NicolasM0**
Reported height test ratio was always equal to the position test ratio.

I fixed that using the correct index in _gps_pos_test_ratio vector.

I also add all height sources in this status.

Test: On the ground with baro as main height source, when I blow on my autopilot, the height test ratio increase.

### CODE REVIEW

File: EKF/control.cpp [hunk 1]
```diff
@@ -1071,6 +1071,13 @@ void Ekf::controlHeightFusion()
 	}
 
 	if (fuse_height) {
+
+		// Reset all height tests ratio
+		_gps_pos_test_ratio(1) = 0;
+		_ev_pos_test_ratio(1) = 0;
+		_baro_hgt_test_ratio(1) = 0;
+		_rng_hgt_test_ratio(1) = 0;
```
Comment (bresch) [inline EKF/control.cpp:1079]:
```suggestion
		resetHeightTestRatios();
```
Could you please put them in a small function like that?

**Review by bresch — CHANGES_REQUESTED** [review]

File: EKF/ekf_helper.cpp [hunk 2]
```diff
@@ -910,7 +910,11 @@ void Ekf::get_innovation_test_status(uint16_t &status, float &mag, float &vel, f
 	pos = math::max(sqrtf(_gps_pos_test_ratio(0)), sqrtf(_ev_pos_test_ratio(0)));
 
 	// return the vertical position innovation test ratio
-	hgt = sqrtf(_gps_pos_test_ratio(0));
+	hgt = sqrtf(math::max(math::max(_gps_pos_test_ratio(1),
```
Comment (dagar) [inline EKF/ekf_helper.cpp:913]: Shouldn't we be returning the corresponding test ratio for the current height mode?

Comment (NicolasM0) [reply EKF/ekf_helper.cpp:913]: To be sure this return the current height ratio, I added a reset of all height ratios before calculating the current one .

I can use `ratio._control_status.flags.*_hgt` here to find the correct height ratio if you think that is better but I thought that is easier to read like that and more robust to future changes like that.

Comment (dagar) [reply EKF/ekf_helper.cpp:913]: I have a minor preference for being lazier (not blanket resetting everything every iteration regardless of use), but I think it's fine either way here.

Comment (NicolasM0) [reply EKF/ekf_helper.cpp]: here is a new implementation without useless resets

**Review by bresch — APPROVED** [review]

**Review by dagar — APPROVED** [review]

**Pull request merged**